src/Functions/extractAllGroups.h

```diff
@@ -173,10 +173,11 @@ class FunctionExtractAllGroups : public IFunction
                         all_matches.push_back(matched_groups[group]);
 
                     /// Additional limit to fail fast on supposedly incorrect usage.
-                    static constexpr size_t MAX_GROUPS_PER_ROW = 1000000;
-
-                    if (all_matches.size() > MAX_GROUPS_PER_ROW)
-                        throw Exception(ErrorCodes::TOO_LARGE_ARRAY_SIZE, "Too large array size in the result of function {}", getName());
+                    static constexpr size_t MAX_MATCHES_PER_ROW = MAX_GROUPS_COUNT * MAX_GROUPS_COUNT;
```

Comment:
Non-obvious - why squared?

Reply:
No particular reason, could be anything else. 1M matches per row seemed too high, 128 seemed too low.

Reply:
updated it to be 1000

Reply:
Ok, let's fix typo and it's ready.